core/src/main/java/org/everit/email/queue/internal/EmailQueueBulkEmailSender.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2011 Everit Kft. (http://www.everit.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.everit.email.queue.internal;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.List;
+
+import org.everit.email.Email;
+import org.everit.email.queue.CreatePassOnJobParam;
+import org.everit.email.queue.schema.qdsl.QQueueMail;
+import org.everit.email.sender.BulkEmailSender;
+import org.everit.email.sender.EmailSender;
+import org.everit.email.store.EmailStore;
+import org.everit.persistence.querydsl.support.QuerydslSupport;
+import org.everit.transaction.propagator.TransactionPropagator;
+
+import com.mysema.query.sql.SQLQuery;
+import com.mysema.query.sql.dml.SQLDeleteClause;
+import com.mysema.query.sql.dml.SQLInsertClause;
+import com.mysema.query.types.Projections;
+
+/**
+ * Email Queue Bulk email sender implementation.
+ */
+public class EmailQueueBulkEmailSender implements BulkEmailSender {
+
+  /**
+   * Queue email data holder.
+   */
+  public static class QueueEmailDTO {
+
+    public long queueEmailId;
+
+    public long storedEmailId;
+  }
+
+  private EmailSender emailSender;
+
+  private EmailStore emailStore;
+
+  private QuerydslSupport querydslSupport;
+
+  private TransactionPropagator transactionPropagator;
+
+  /**
+   * Simple constructor.
+   *
+   * @param emailSender
+   *          an {@link EmailSender} instance.
+   * @param emailStore
+   *          an {@link EmailStore} instance.
+   * @param querydslSupport
+   *          a {@link QuerydslSupport} instance.
+   * @param transactionPropagator
+   *          a {@link TransactionPropagator} instance.
+   */
+  public EmailQueueBulkEmailSender(final EmailSender emailSender, final EmailStore emailStore,
+      final QuerydslSupport querydslSupport, final TransactionPropagator transactionPropagator) {
+    this.emailSender = emailSender;
+    this.emailStore = emailStore;
+    this.querydslSupport = querydslSupport;
+    this.transactionPropagator = transactionPropagator;
+  }
+
+  @Override
+  public void close() {
+    // do nothing
+  }
+
+  /**
+   * Creates pass on job that send mails.
+   */
+  public Runnable createPassOnJob(final CreatePassOnJobParam param) {
+    return () -> {
+      transactionPropagator.required(() -> {
+        List<QueueEmailDTO> queuedEmails = selectAndLockQueueEmail(param);
```

Comment:
I would call QueuedEmailDTO and everywhere else I would take there the "d" letter